src/App.tsx

```diff
@@ -1,24 +1,332 @@
+        setErrorMessage('Unable to delete completed todos');
+      })
+      .finally(() => setLoadingTodoId([]));
+  }, [todos]);
+
+  const toggleTodoStatus = async (todoId: number, completed: boolean) => {
+    try {
+      setLoadingTodoId([todoId]);
+      await todoService.updatePostStatus(todoId, completed);
+
+      setTodos(currentTodos => currentTodos.map(
+        todo => (todoId === todo.id ? { ...todo, completed } : todo),
+      ));
+    } catch {
+      setErrorMessage('Unable to update a todo');
+    } finally {
+      setLoadingTodoId([]);
+    }
+  };
+
+  const handleToggleTodosAll = async (completed: boolean) => {
+    const todosIds = todos
+      .filter(todo => todo.completed !== completed)
+      .map(todo => todo.id);
+
+    try {
+      setLoadingTodoId(todosIds);
+
+      const updatedTodos = todos.map(todo => ({
+        ...todo,
+        completed,
+      }));
+
+      setTodos(updatedTodos);
+
+      await Promise.all(
+        todos.map(todo => todoService.updatePostStatus(todo.id, completed)),
+      );
+    } catch {
+      setErrorMessage('Unable to update a todos');
+    } finally {
+      setLoadingTodoId([]);
+    }
+  };
+
+  const handleEditTodo = async (todoId: number, newTitle: string) => {
+    if (!newTitle) {
+      setErrorMessage('Title can`t be empty');
+
+      return;
+    }
+
+    try {
+      setLoadingTodoId([todoId]);
+      const updatedTodo = await todoService.updateTodoTitle(todoId, newTitle);
+
+      setTodos(currentTodos => currentTodos.map(
+        todo => (todoId === todo.id ? updatedTodo : todo),
+      ));
+    } catch {
+      setErrorMessage('Title can`t be empty');
+    } finally {
+      setLoadingTodoId([]);
+    }
+  };
+
+  useEffect(() => {
+    if (!disableInput && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [disableInput]);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  useEffect(() => {
+    todoService.getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load a todo'));
+
+    setTimeout(() => {
+      setVisible(true);
+    }, 1000);
+  }, []);
+
   if (!USER_ID) {
-    return <UserWarning />;
+    return (
+      <UserWarning />
+    );
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div
+        className={classNames('todoapp__content', {
+          visible,
+        })}
+      >
+        <header className="todoapp__header">
+          {todos.length > 0 && (
```

Comment:
you can convert it to boolean like this:
```suggestion
          {!!todos.length && (
```